### PR TITLE
CORDA-2786: Certificates directory is only required for node startup

### PR DESCRIPTION
**Pull request by lockathan**
Certificates directory should only be required on node startup. For all other cases, it is optional. This was breaking initial registration in particular, which is usually responsible for creating the certificates directory.

initialiseAndRun is called from all of the different CLI commands, including but not limited to generate-node-info and initial-registration.

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt [hunk 1]
```diff
@@ -151,8 +151,9 @@ open class NodeStartup : NodeStartupLogging {
         val configuration = cmdLineOptions.parseConfiguration(rawConfig).doIfValid { logRawConfig(rawConfig) }.doOnErrors(::logConfigurationErrors).optional ?: return ExitCodes.FAILURE
        
         // Step 6. Check if we can access the certificates directory
-        if (!canReadCertificatesDirectory(configuration.certificatesDirectory, configuration.devMode)) return ExitCodes.FAILURE
-
+        if (requireCertificates) {
```
Comment (chrisr3) [inline node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt]: I don't like the nested `if`. What's wrong with `and`?

File: node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt [hunk 2]
```diff
@@ -111,7 +111,7 @@ open class NodeStartupCli : CordaCliWrapper("corda", "Runs a Corda Node") {
             else -> startup.initialiseAndRun(cmdLineOptions, object : RunAfterNodeInitialisation {
                 val startupTime = System.currentTimeMillis()
                 override fun run(node: Node) = startup.startNode(node, startupTime)
-            })
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt]: If we're requesting changes, can you add the named parameter for the `true`.

Comment (lockathan) [reply node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt]: Hi Shams. Sorry, I don't understand this comment? I do pass the value "true" in here. Do you mean that I should name the parameter at the point where it is passed in?

Comment (shamsasari) [reply node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt]: Yes, that's what I meant. Apologies for not being clear. `requireCertificates = true`

File: node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt (hunk 1 above)
Comment (lockathan) [reply node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt]: Fixed - please have a look.

File: node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt (hunk 2 above)
Comment (lockathan) [reply node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt]: Fixed - please have a look.

File: node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt [hunk 3]
```diff
@@ -151,8 +151,8 @@ open class NodeStartup : NodeStartupLogging {
         val configuration = cmdLineOptions.parseConfiguration(rawConfig).doIfValid { logRawConfig(rawConfig) }.doOnErrors(::logConfigurationErrors).optional ?: return ExitCodes.FAILURE
        
         // Step 6. Check if we can access the certificates directory
-        if (!canReadCertificatesDirectory(configuration.certificatesDirectory, configuration.devMode)) return ExitCodes.FAILURE
-
+        if (requireCertificates && !canReadCertificatesDirectory(configuration.certificatesDirectory, configuration.devMode)) return ExitCodes.FAILURE
+
```
Comment (chrisr3) [inline node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt]: _AAAGH! Evil Whitespace!_ (And the unused `import` too...)

**Pull request merged**